packages/core/src/state/merge/mergeDeepCopy.ts

```diff
@@ -5,15 +5,25 @@ import isMergeableObject from './isMergeable';
 /**
  * Deep merge two objects or arrays. Uses static merge function if exists.
  */
-export default function mergeDeepCopy<T1, T2>(target: T1, source: T2): T1 & T2 {
+export default function mergeDeepCopy<T1, T2>(
+  target: T1,
+  source: T2,
+  targetMeta?: any,
+  sourceDate?: any,
+): T1 & T2 {
   const sourceIsArray = Array.isArray(source);
   const targetIsArray = Array.isArray(target);
   const sourceAndTargetTypesMatch = sourceIsArray === targetIsArray;
   const Static: StaticType<T2> = source && (source as any).constructor;
 
   if (target && Static && isMergeable(Static)) {
     if (isMergeable((target as any).constructor)) {
-      return Static.merge(target, source) as any;
+      // second argument takes priority over first
+      // if either of these is undefined, it resolves to 'false' which
+      // means we fallback to 'newer' (source) takes priority
+      return (targetMeta?.date > sourceDate
+        ? Static.merge(source, target)
```

Comment:
perhaps it is worthwhile to send more information from the resolution (maybe an object with date and which manager fired the action or the entire action) and let the a default implementation of merge control the behavior.

i can't think of an example where this would be useful though so I don't think we should worry about it.

Reply:
Interesting idea. I prefer to minimize public interfaces as well as coupling of concerns without a clear need. If/when this need is realized the interface for such a thing will be more clear. It is easier to add than change.

I'll keep this in mind though!